extensions/kafka/src/main/java/com/hazelcast/jet/kafka/KafkaDataConnection.java

```diff
@@ -189,15 +189,23 @@ public <K, V> KafkaProducer<K, V> getProducer(@Nullable String transactionalId)
     public <K, V> KafkaProducer<K, V> getProducer(
             @Nullable String transactionalId,
             @Nonnull Properties properties) {
-        if (getConfig().isShared() && !properties.isEmpty()) {
+        Properties configProperties = getConfig().getProperties();
+
+        boolean inputPropsAreSubsetOfConfigProps = inputPropsAreSubsetOfConfigProps(properties);
+        if (getConfig().isShared() && !properties.isEmpty() && !inputPropsAreSubsetOfConfigProps) {
```

Comment:
The error message is now misleading, because you still _must_ provide the `keyFormat`/`valueFormat` when creating the SQL mapping AFAICT. Also, providing only the `*.deserializer` options for the DC and key/value formats for the mapping isn't enough either (they aren't considered a subset). This is poor UX, but I think we can live with it in BETA if we improve the error messaging. Maybe something that points the user to set the `*.serializer` options when creating the DC?